File: tags/CPP_RELEASE_1_7_2_0/x10.lib.crt/src/place.cc
#include "x10.h"


X10::Place::Place(x10_place_t id) :
    __id(id) {} 
  
X10::Place::operator x10_place_t() const
{
  return __id;
}

X10::Place::operator x10_place_t() 
{
  return __id;
}

X10::Place::Place(const Place& p) :
  __id(p.__id) {}

const X10::Place::Place& X10::Place:: operator=(const Place& p)
{
  __id = p.__id;
  return *this;
}

const bool X10::Place::operator== (const X10::Place::Place& p)const
{
  return __id == p.__id;
}
  
  

File: apgas/src/apgas/Place.java
/*
 *  This file is part of the X10 project (http://x10-lang.org).
 *
 *  This file is licensed to You under the Eclipse Public License (EPL);
 *  You may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *      http://www.opensource.org/licenses/eclipse-1.0.php
 *
 *  (C) Copyright IBM Corporation 2006-2015.
 */

package apgas;

import java.io.Serializable;

/**
 * The {@link Place} class represents an APGAS place.
 */
public class Place implements Serializable, Comparable<Place> {
  private static final long serialVersionUID = -7210312031258955537L;

  /**
   * The integer ID of this place.
   */
  public final int id;

  /**
   * Constructs a {@link Place} with the specified ID.
   * <p>
   * Prefer {@link Constructs#place(int)} to avoid constructing unnecessary
   * {@link Place} objects.
   *
   * @param id
   *          the desired place ID
   */
  public Place(int id) {
    this.id = id;
  }

  @Override
  public String toString() {
    return "place(" + id + ")";
  }

  @Override
  public boolean equals(Object that) {
    return that instanceof Place ? id == ((Place) that).id : false;
  }

  @Override
  public int hashCode() {
    return Integer.hashCode(id);
  }

  @Override
  public int compareTo(Place o) {
    return Integer.compare(id, o.id);
  }
}
